test/tools/ossfuzz/lpsolver/ConstraintGenerator.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+	This file is part of solidity.
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+#include <test/tools/ossfuzz/lpsolver/ConstraintGenerator.h>
+
+using namespace std;
+using namespace solidity::test::fuzzer::lpsolver;
+
+ConstraintGenerator::ConstraintGenerator(unsigned int _seed)
+{
+	prng = make_shared<RandomEngine>(_seed);
+}
+
+string ConstraintGenerator::generate()
+{
+	string constraint;
+	for (int i = 0; i < numConstraints(); i++)
+	{
+		// First entry is always constraint type. If it is equal to "1", it is an equality constraint
+		// otherwise an less-than-equal constraint.
+		constraint += to_string(zeroOrOne());
+		for (int j = 0; j < numFactors(); j++)
+			if (bernoulliDist(s_piecewiseConstantProb))
```

Comment:
Why this distinction?

Reply:
Since the random integer generator (used for generating constant term and factors for variables in linear expressions) uses uniform random distribution over the range [INT_MIN, INT_MAX], the probability of zero is roughly `1/(INT_MAX - INT_MIN)` which is minuscule. If it weren't for this distinction, linear expressions that do not feature certain variables would be very rare because by convention the integer vector `{i0, i1, i2,...,in}` translates to

```
i1*x1 + i2*x2 + ... + in*xn <= i0
```

If `ik` is rarely zero, then it features almost always in a constraint which we may not really desire.

Reply:
Having said that the argument for `1` and `-1` in the piecewise range is weaker. Basically, I am not sure if they are special enough to be corner cases but I thought they wouldn't hurt :)